cfme/utils/ssh.py

```diff
@@ -842,3 +851,48 @@ def keygen():
     with _ssh_pubkey_file.open('w') as f:
         f.write("{} {} {}\n".format(pub.get_name(), pub.get_base64(),
             'autogenerated cfme_tests key'))
+
+
+def connect_ssh(vm, creds,
+                num_sec=CONNECT_RETRIES_TIMEOUT,
+                connect_timeout=CONNECT_TIMEOUT,
+                delay=CONNECT_SSH_DELAY):
+    """
+    This function attempts to connect all the IPs of the vm in several
+    rounds. After each round it will delay.
+    Once connective IP is found, it will return SSHClient with established
+    connection.
+    """
+
+    msg = "The num_sec is smaller then connect_timeout. This looks like a bug."
+    assert num_sec > connect_timeout, msg
+
+    timestamp = time.monotonic()
+    # Note we do need to be sure we get the fresh list of IPs for every cycle.
+    # The provider may be updating it.
+    for ip in _trying_fresh_ips(vm, delay=delay, rounds=-1):
+        # Timeout handling.
+        total_duration = (time.monotonic() - timestamp)
+        if total_duration > num_sec:
+            raise TimedOutError(f"None of the ips of vm {vm} were found "
+                f"connectible after trying each in {total_duration:.0f} seconds "
+                "total.")
+
+        try:
+            client = SSHClient(hostname=ip, username=creds.principal, password=creds.secret,
+                               use_check_port=False, timeout=connect_timeout)
+            client.connect()
+        except Exception as ex:
+            logger.info(f"Failed to connect with SSH to {ip}: {ex}")
+            continue
+        else:
+            logger.info(f"SSH connected to IP {ip}")
+
+            # This command execution check is probably redundant.
+            result = client.run_command("true")
```

Comment:
Agreed this is redundant, I'd remove it, no need for the extra load of making a token command call.

Reply:
Removed.